pkg/raft/raft.go

```diff
@@ -2647,11 +2647,11 @@ func (r *raft) loadState(state pb.HardState) {
 	r.setLeadEpoch(state.LeadEpoch)
 }
 
-// pastElectionTimeout returns true if r.electionElapsed is greater
-// than or equal to the randomized election timeout in
-// [electiontimeout, 2 * electiontimeout - 1].
-func (r *raft) pastElectionTimeout() bool {
-	return r.electionElapsed >= r.randomizedElectionTimeout
+// atRandomizedElectionTimeout returns true if r.electionElapsed modulo the
+// r.randomizedElectionTimeout is equal to 0. This means that at every
+// r.randomizedElectionTimeout period, this method will return true once.
+func (r *raft) atRandomizedElectionTimeout() bool {
```

Comment:
The caller of this doesn't check `promotable()` anymore, so what happens in the following case?

1. tick `randomizedElectionTimeout`: we call `hup`, but `promotable()` is false, so we don't campaign
2. tick `randomizedElectionTimeout+1`: we became promotable, but we will have to wait out another `randomizedElectionTimeout` to campaign

Is this intended?